src/app/components/core/control-dispatcher/list-control-dispatcher/control-tasks-list/control-tasks-list.component.ts

```diff
@@ -0,0 +1,200 @@
+import { Component, OnInit, ViewChild, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
+import {
+  AlertService,
+  SharedService,
+  ProgressBarService,
+  SchedulesService,
+  ConfigurationService
+} from '../../../../../services';
+import { ControlDispatcherService } from '../../../../../services/control-dispatcher.service';
+import { ConfirmationDialogComponent } from '../../confirmation-dialog/confirmation-dialog.component';
+import { DialogService } from '../../confirmation-dialog/dialog.service';
+import { orderBy, isEmpty } from 'lodash';
+
+@Component({
+  selector: 'app-control-tasks-list',
+  templateUrl: './control-tasks-list.component.html',
+  styleUrls: ['./control-tasks-list.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+
+})
+export class ControlTasksListComponent implements OnInit {
+  controlScripts: any = [];
+  script;
+  @ViewChild('confirmationDialog') confirmationDialog: ConfirmationDialogComponent;
+  controlTask;
+  constructor(private controlService: ControlDispatcherService,
+    private schedulesService: SchedulesService,
+    private configurationService: ConfigurationService,
+    private alertService: AlertService,
+    private dialogService: DialogService,
+    public sharedService: SharedService,
+    private cd: ChangeDetectorRef,
+    private ngProgress: ProgressBarService) { }
+
+  ngOnInit(): void {
+    this.getControlScripts();
+  }
+
+  setScript(script: any) {
+    this.script = script;
+  }
+
+  openModal(id: string) {
+    this.dialogService.open(id);
+  }
+
+  closeModal(id: string) {
+    this.dialogService.close(id);
+  }
+
+  getControlScripts() {
+    /** request started */
+    this.ngProgress.start();
+    this.controlService.fetchControlServiceScripts()
+      .subscribe((data: any) => {
+        this.ngProgress.done();
+        this.controlScripts = orderBy(data.scripts, 'name');
+        this.controlScripts = this.controlScripts.filter(c => !isEmpty(c.schedule))
+        this.cd.detectChanges();
+      }, error => {
+        /** request completed */
+        this.ngProgress.done();
+        if (error.status === 0) {
+          console.log('service down ', error);
+        } else {
+          this.alertService.error(error.statusText);
+        }
+      });
+  }
+
+  deleteControlSchedule(script) {
+    const id = script.schedule.id;
+    /** request started */
+    this.ngProgress.start();
+    // disable schedule
+    this.schedulesService.disableSchedule(id)
+      .subscribe(() => {
+        // delete schedule
+        this.schedulesService.deleteSchedule(id)
```

Comment:
I think delete API will disable the schedule.

Reply:
not happening, it asked to disable the schedule first.